ext/src/helpers/nogvl.rs

```diff
@@ -0,0 +1,29 @@
+use std::{ffi::c_void, mem::MaybeUninit, ptr::null_mut};
+
+use rb_sys::rb_thread_call_without_gvl;
+
+unsafe extern "C" fn call_without_gvl<F, R>(arg: *mut c_void) -> *mut c_void
+where
+    F: FnMut() -> R,
+    R: Sized,
+{
+    let arg = arg as *mut (&mut F, &mut MaybeUninit<R>);
+    let (func, result) = unsafe { &mut *arg };
+    result.write(func());
+
+    null_mut()
+}
+
+pub fn nogvl<F, R>(mut func: F) -> R
+where
+    F: FnMut() -> R,
+    R: Sized,
+{
+    let result = MaybeUninit::uninit();
+    let arg_ptr = &(&mut func, &result) as *const _ as *mut c_void;
+
+    unsafe {
+        rb_thread_call_without_gvl(Some(call_without_gvl::<F, R>), arg_ptr, None, null_mut());
```

Comment:
re `None` ubf, [according to the doc](https://github.com/ruby/ruby/blob/3fe2f2689f0074a4c880218ca08cb10137880760/thread.c#L1577-L1579):

> Be careful to use rb_thread_call_without_gvl(). If you don't provide proper ubf(), your program will not stop for Control+C or other shutdown events.

For compilation, that seems fine: (1) I don't know of a way to interrupt compilation (unsure) (2) that's no different than today's behaviour.